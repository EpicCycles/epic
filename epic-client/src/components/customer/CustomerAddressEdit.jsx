import React from "react";
import {Icon} from "semantic-ui-react";
import {createEmptyModelWithDefaultFields, getModelKey, isModelValid, updateModel} from "../app/model/helpers/model";
import {customerAddressFields} from "../app/model/helpers/fields";
import EditModelRow from "../app/model/EditModelRow";
import * as PropTypes from "prop-types";
import {isItAnObject} from "../../helpers/utils";
import ModelEditIcons from "../app/model/ModelEditIcons";

const initialState = {};

class CustomerAddressEdit extends React.Component {
    state = initialState;

    componentWillMount() {
        this.setState(this.deriveStateFromProps());
    };

    componentDidUpdate(prevProps) {
        if (this.props.customerAddress !== prevProps.customerAddress) this.deriveStateFromProps();
    }

    deriveStateFromProps = () => {
        if (isItAnObject(this.props.customerAddress)) {
           return {customerAddress: this.props.customerAddress};
        } else {
            let customerAddress =  createEmptyModelWithDefaultFields(customerAddressFields);
            customerAddress.customer = this.props.customerId;
            return { customerAddress }
        }
    };

    handleInputChange = (fieldName, input) => {
        const updatedCustomerAddress = updateModel(this.state.customerAddress, customerAddressFields, fieldName, input);
        this.setState({ customerAddress: updatedCustomerAddress });
    };

    onClickReset = () => {
        this.setState(this.deriveStateFromProps());
    };
    render() {
        const { customerAddress } = this.state;
        const {saveCustomerAddress, deleteCustomerAddress} = this.props;
        const componentKey = getModelKey(customerAddress);
        const rowClass = (customerAddress && customerAddress.error) ? "grid-row error" : "grid-row";
        return <div className={rowClass} key={`row${componentKey}`}>
            <EditModelRow
                model={customerAddress}
                persistedModel={this.props.customerAddress}
                modelFields={customerAddressFields}
                onChange={this.handleInputChange}
            />
            <div key={`date_${componentKey}`} className="grid-item">
                {(customerAddress && customerAddress.add_date) ?
                    <nobr id={`comment_td_${componentKey}`}>Added on {customerAddress.add_date.substring(0, 10)}, last updated on {customerAddress.upd_date.substring(0, 10)}</nobr>
                    : <nobr id={`comment_td_${componentKey}`}>Add a new address</nobr>
                }
            </div>
            <div
                key={`actions_td_${componentKey}`}
                className="grid-item align_center"
            >
                 <ModelEditIcons
                     componentKey={componentKey}
                     model={customerAddress}
                     modelSave={saveCustomerAddress}
                     modelDelete={deleteCustomerAddress}
                     modelReset={this.onClickReset}
                 />
            </div>
        </div>;
    }
}

CustomerAddressEdit.defaultProps = {
    customerAddress: {},
};
CustomerAddressEdit.propTypes = {
    customerAddress: PropTypes.object,
    deleteCustomerAddress: PropTypes.func.isRequired,
    saveCustomerAddress: PropTypes.func.isRequired,
    customerId: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ])
};
export default CustomerAddressEdit;

