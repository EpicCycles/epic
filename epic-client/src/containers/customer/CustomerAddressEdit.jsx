import React from "react";
import {Icon} from "semantic-ui-react";
import FormTextInput from "../../common/FormTextInput";
import {validatePostcodeFormat} from "../../helpers/utils";

const initialState = {
    address1: '',
    address2: '',
    address3: '',
    address4: '',
    postcode: '',
    address1Error: '',
    postcodeError: '',
    isChanged: false,
    isValid: true
};

class CustomerAddressEdit extends React.Component {
    state = initialState;

    componentWillMount() {
        if (this.props.customerAddress) {
            this.setState({
                address1: this.props.customerAddress.address1,
                address2: this.props.customerAddress.address2,
                address3: this.props.customerAddress.address3,
                address4: this.props.customerAddress.address4,
                postcode: this.props.customerAddress.postcode
            });
        }
    };

    validateCustomerAddressData = (address1, address2, address3, address4, postcode) => {
        let isChanged = false;
        let isValid = true;
        let address1Error = "";
        let postcodeError = "";

        if (this.props.customerAddress) {
            if (this.props.customerAddress.address1 !== address1
                || this.props.customerAddress.address2 !== address2
                || this.props.customerAddress.address3 !== address3
                || this.props.customerAddress.address4 !== address4
                || this.props.customerAddress.postcode !== postcode) {
                isChanged = true;
            }
        } else {
            isChanged = (address1 || address2 || address3 || address4 || postcode);
        }

        if (isChanged) {
            if (!(address1)) address1Error = "At least 1 line of address must be provided";
            if (!(postcode)) postcodeError = "Postcode must be provided";
        }

        if (address1Error || postcodeError) isValid = false;

        this.setState({
            address1: address1,
            address2: address2,
            address3: address3,
            address4: address4,
            postcode: postcode,
            address1Error: address1Error,
            postcodeError: postcodeError,
            isChanged: isChanged,
            isValid: isValid
        });
    };

    handleInputChange = (fieldName, input) => {
        let address1 = this.state.address1;
        let address2 = this.state.address2;
        let address3 = this.state.address3;
        let address4 = this.state.address4;
        let postcode = this.state.postcode;
        if (fieldName.startsWith('address1')) address1 = input;
        if (fieldName.startsWith('address2')) address2 = input;
        if (fieldName.startsWith('address3')) address3 = input;
        if (fieldName.startsWith('address4')) address4 = input;
        if (fieldName.startsWith('postcode')) postcode = input;

        this.validateCustomerAddressData(address1, address2, address3, address4, postcode);
    };

    handleInputClear = (fieldName) => {
        let address1 = this.state.address1;
        let address2 = this.state.address2;
        let address3 = this.state.address3;
        let address4 = this.state.address4;
        let postcode = this.state.postcode;
        if (fieldName.startsWith('removeaddress1')) address1 = this.props.customerAddress ? this.props.customerAddress.address1 : '';
        if (fieldName.startsWith('removeaddress2')) address2 = this.props.customerAddress ? this.props.customerAddress.address2 : '';
        if (fieldName.startsWith('removeaddress3')) address3 = this.props.customerAddress ? this.props.customerAddress.address3 : '';
        if (fieldName.startsWith('removeaddress4')) address4 = this.props.customerAddress ? this.props.customerAddress.address4 : '';
        if (fieldName.startsWith('removepostcode')) postcode = this.props.customerAddress ? this.props.customerAddress.postcode : '';

        this.validateCustomerAddressData(address1, address2, address3, address4, postcode);
    };

    onClickReset = () => {
        this.setState(initialState);
    };

    saveOrCreateCustomerAddress = () => {
        if (this.state.postcode && !validatePostcodeFormat(this.state.postcode)) {
           this.setState({ postcodeError: "Invalid Postcode", isValid: false});
           return false;
        }

        if (this.props.customerAddress) {
            let addressToSave = this.props.customerAddress;
            addressToSave.address1 = this.state.address1;
            addressToSave.address2 = this.state.address2;
            addressToSave.address3 = this.state.address3;
            addressToSave.address4 = this.state.address4;
            addressToSave.postcode = this.state.postcode;
            this.props.saveCustomerAddress(addressToSave);
        }
        else {
            const newAddress = {
                customer: this.props.customerId,
                address1: this.state.address1,
                address2: this.state.address2,
                address3: this.state.address3,
                address4: this.state.address4,
                postcode: this.state.postcode
            };
            this.props.saveCustomerAddress(newAddress);
            this.setState(initialState);
        }
    };

    onClickDelete = () => {
        if (this.props.customerAddress) {
            let addressToSave = this.props.customerAddress;
            this.props.deleteCustomerAddress(addressToSave.id);
        } else {
            this.setState(initialState);
        }
    };

    render() {
        const { address1, address2, address3, address4, postcode, isChanged, isValid, address1Error, postcodeError } = this.state;
        const { customerAddress } = this.props;
        const keyValue = (customerAddress && customerAddress.id) ? customerAddress.id : "new";
        const componentContext = customerAddress ? customerAddress.id : 'newAddress';
        return <tr id={componentContext}>
            <td>
                <FormTextInput
                    placeholder="Address line 1"
                    id={`address1-input_${componentContext}`}
                    className="column full"
                    value={address1}
                    fieldName={`address1_${componentContext}`}
                    onChange={this.handleInputChange}
                    onClick={this.handleInputClear}
                    error={address1Error}
                />
            </td>
            <td>
                <FormTextInput
                    placeholder="Address line 2"
                    id={`address2-input_${componentContext}`}
                    className="column full"
                    value={address2}
                    fieldName={`address2_${componentContext}`}
                    onChange={this.handleInputChange}
                    onClick={this.handleInputClear}
                />
            </td>
            <td>
                <FormTextInput
                    placeholder="Address line 3"
                    id={`address3-input_${componentContext}`}
                    className="column full"
                    value={address3}
                    fieldName={`address3_${componentContext}`}
                    onChange={this.handleInputChange}
                    onClick={this.handleInputClear}
                />
            </td>
            <td>
                <FormTextInput
                    placeholder="Address line 4"
                    id={`address4-input_${componentContext}`}
                    className="column full"
                    value={address4}
                    fieldName={`address4_${componentContext}`}
                    onChange={this.handleInputChange}
                    onClick={this.handleInputClear}
                />
            </td>
            <td>
                <FormTextInput
                    placeholder="Postcode"
                    id={`postcode-input_${componentContext}`}
                    className="column full"
                    value={postcode}
                    fieldName={`postcode_${componentContext}`}
                    onChange={this.handleInputChange}
                    onClick={this.handleInputClear}
                    error={postcodeError}
                />
            </td>
            <td>
                {(customerAddress && customerAddress.add_date) ?
                    <span>Added on {customerAddress.add_date.substring(0, 10)}, last updated on {customerAddress.upd_date.substring(0, 10)}</span>
                    : <span>Add a new address</span>
                }
            </td>
            <td>
                  <span id={`actions${keyValue}`}>
                      {isChanged &&
                      <Icon id={`reset-address${keyValue}`} name="undo"
                            onClick={this.onClickReset} title="Reset Address details"
                      />
                      }
                      {isChanged &&
                      <Icon id={`accept-address${keyValue}`} name="check" disabled={!isValid}
                            onClick={this.saveOrCreateCustomerAddress}
                            title="Confirm changes"/>
                      }
                      {(address1 || address2 || address3 || address4 || postcode) &&
                      <Icon id={`delete-address${keyValue}`} name="delete"
                            onClick={this.onClickDelete}
                            title="Delete Address Number"/>
                      }
                </span>
            </td>
        </tr>;
    }
}

export default CustomerAddressEdit;

