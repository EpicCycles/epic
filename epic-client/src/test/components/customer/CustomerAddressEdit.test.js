import React from 'react';
import CustomerAddressEdit from "../../../components/customer/CustomerAddressEdit";
// props customerAddress: {address1/2/3/4, postcode}, deleteCustomerAddress,saveCustomerAddress
it("renders correctly with a passed customer address", () => {
    const customerAddress = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6
    };
    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
        />
    );
    expect(component).toMatchSnapshot();
    expect(component.state('address1')).toBe("line one");
    expect(component.state('address2')).toBe("line Ywo");
    expect(component.state('address3')).toBe("line Three");
    expect(component.state('address4')).toBe("line Four");
    expect(component.state('postcode')).toBe("xxxyyy");
    expect(component.state('address1Error')).toBe("");
    expect(component.state('postcodeError')).toBe("");
    expect(component.state('isChanged')).toBeFalsy();
    expect(component.state('isValid')).toBeTruthy();
});
it("renders correctly with a new customer address", () => {
    const customerAddress = {};
    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
        />
    );
    expect(component).toMatchSnapshot();
    expect(component.state('address1')).toBe("");
    expect(component.state('address2')).toBe("");
    expect(component.state('address3')).toBe("");
    expect(component.state('address4')).toBe("");
    expect(component.state('postcode')).toBe("");
    expect(component.state('address1Error')).toBe("");
    expect(component.state('postcodeError')).toBe("");
    expect(component.state('isChanged')).toBeFalsy();
    expect(component.state('isValid')).toBeTruthy();
});
it("processes input changes and sets errors as appropriate for an existing customer", () => {
    const customerAddress = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6
    };
    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
        />
    );
    component.instance().handleInputChange("address1_123", 'new line 1');
    expect(component.state('isChanged')).toBeTruthy();
    expect(component.state('address1')).toBe("new line 1");
    expect(component.state('isValid')).toBeTruthy();

    component.instance().handleInputChange("address1_123", '');
    expect(component.state('isChanged')).toBeTruthy();
    expect(component.state('address1')).toBe("");
    expect(component.state('isValid')).toBeFalsy();
    expect(component.state('address1Error')).toBe("At least 1 line of address must be provided");

    component.instance().handleInputChange("address1_123", 'new line 21');
    component.instance().handleInputChange("address2_123", 'new line 22');
    component.instance().handleInputChange("address3_123", 'new line 23');
    component.instance().handleInputChange("address4_123", 'new line 24');
    component.instance().handleInputChange("postcode_123", 'new postcode');
    expect(component.state('address1')).toBe("new line 21");
    expect(component.state('address2')).toBe("new line 22");
    expect(component.state('address3')).toBe("new line 23");
    expect(component.state('address4')).toBe("new line 24");
    expect(component.state('postcode')).toBe("new postcode");
    expect(component.state('isChanged')).toBeTruthy();
    expect(component.state('isValid')).toBeTruthy();

    component.instance().handleInputChange("postcode_123", '');
    expect(component.state('isChanged')).toBeTruthy();
    expect(component.state('postcode')).toBe("");
    expect(component.state('isValid')).toBeFalsy();
    expect(component.state('postcodeError')).toBe("Postcode must be provided");

    component.instance().handleInputChange("address1_123", customerAddress.address1);
    component.instance().handleInputChange("address2_123", customerAddress.address2);
    component.instance().handleInputChange("address3_123", customerAddress.address3);
    component.instance().handleInputChange("address4_123", customerAddress.address4);
    component.instance().handleInputChange("postcode_123", customerAddress.postcode);
    expect(component.state('address1Error')).toBe("");
    expect(component.state('postcodeError')).toBe("");
    expect(component.state('isChanged')).toBeFalsy();
    expect(component.state('isValid')).toBeTruthy();
});
it("processes input changes and sets errors as appropriate for a new customer", () => {
    const customerAddress = {};
    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
        />
    );
    component.instance().handleInputChange("address1_123", 'new line 21');
    component.instance().handleInputChange("address2_123", 'new line 22');
    component.instance().handleInputChange("address3_123", 'new line 23');
    component.instance().handleInputChange("address4_123", 'new line 24');
    component.instance().handleInputChange("postcode_123", 'new postcode');
    expect(component.state('address1')).toBe("new line 21");
    expect(component.state('address2')).toBe("new line 22");
    expect(component.state('address3')).toBe("new line 23");
    expect(component.state('address4')).toBe("new line 24");
    expect(component.state('postcode')).toBe("new postcode");
    expect(component.state('isChanged')).toBeTruthy();
    expect(component.state('isValid')).toBeTruthy();

    component.instance().handleInputChange("address1_123", '');
    expect(component.state('isChanged')).toBeTruthy();
    expect(component.state('address1')).toBe("");
    expect(component.state('isValid')).toBeFalsy();
    expect(component.state('address1Error')).toBe("At least 1 line of address must be provided");

    component.instance().handleInputChange("postcode_123", '');
    expect(component.state('isChanged')).toBeTruthy();
    expect(component.state('postcode')).toBe("");
    expect(component.state('isValid')).toBeFalsy();
    expect(component.state('postcodeError')).toBe("Postcode must be provided");

    component.instance().handleInputChange("address1_123", "");
    component.instance().handleInputChange("address2_123", "");
    component.instance().handleInputChange("address3_123", "");
    component.instance().handleInputChange("address4_123", "");
    component.instance().handleInputChange("postcode_123", "");
    expect(component.state('isChanged')).toBeFalsy();
    expect(component.state('isValid')).toBeTruthy();
    expect(component.state('address1Error')).toBe("");
    expect(component.state('postcodeError')).toBe("");
});
it("resets back to existing customer when requested", () => {
    const customerAddress = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6
    };
    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
        />
    );
    component.instance().handleInputChange("address1_123", 'new line 21');
    component.instance().handleInputChange("address2_123", 'new line 22');
    component.instance().handleInputChange("address3_123", 'new line 23');
    component.instance().handleInputChange("address4_123", 'new line 24');
    component.instance().handleInputChange("postcode_123", 'new postcode');

    component.instance().onClickReset();

    expect(component.state('isChanged')).toBeFalsy();
    expect(component.state('address1')).toBe("line one");
    expect(component.state('address2')).toBe("line Ywo");
    expect(component.state('address3')).toBe("line Three");
    expect(component.state('address4')).toBe("line Four");
    expect(component.state('postcode')).toBe("xxxyyy");
    expect(component.state('address1Error')).toBe("");
    expect(component.state('postcodeError')).toBe("");
    expect(component.state('isChanged')).toBeFalsy();
    expect(component.state('isValid')).toBeTruthy();
});
it("resets back for new customer when requested", () => {
    const customerAddress = {};
    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
        />
    );
    component.instance().handleInputChange("address1_123", 'new line 21');
    component.instance().handleInputChange("address2_123", 'new line 22');
    component.instance().handleInputChange("address3_123", 'new line 23');
    component.instance().handleInputChange("address4_123", 'new line 24');
    component.instance().handleInputChange("postcode_123", 'new postcode');
    expect(component.state('address1')).toBe("new line 21");
    expect(component.state('address2')).toBe("new line 22");
    expect(component.state('address3')).toBe("new line 23");
    expect(component.state('address4')).toBe("new line 24");
    expect(component.state('postcode')).toBe("new postcode");
    expect(component.state('isChanged')).toBeTruthy();

    component.instance().onClickReset();

    expect(component.state('address1')).toBe("");
    expect(component.state('address2')).toBe("");
    expect(component.state('address3')).toBe("");
    expect(component.state('address4')).toBe("");
    expect(component.state('postcode')).toBe("");
    expect(component.state('isChanged')).toBeFalsy();
    expect(component.state('isValid')).toBeTruthy();
    expect(component.state('address1Error')).toBe("");
    expect(component.state('postcodeError')).toBe("");
});
it("triggers save for an existing customer when requested", () => {
    const customerAddress = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6
    };
    const saveCustomerAddress = jest.fn();
    const customerId = 98;
    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
            saveCustomerAddress={saveCustomerAddress}
            customerId={customerId}
        />
    );
    component.instance().handleInputChange("address1_123", 'new line 21');
    component.instance().handleInputChange("address2_123", 'new line 22');
    component.instance().handleInputChange("address3_123", 'new line 23');
    component.instance().handleInputChange("address4_123", 'new line 24');
    component.instance().handleInputChange("postcode_123", 'new postcode');

    component.instance().saveOrCreateCustomerAddress();

    expect(component.state('isChanged')).toBeTruthy();
    expect(saveCustomerAddress.mock.calls.length).toBe(1);
    expect(saveCustomerAddress.mock.calls[0][0].id).toBe(customerAddress.id);
    expect(saveCustomerAddress.mock.calls[0][0].address1).toBe("new line 21");
    expect(saveCustomerAddress.mock.calls[0][0].address2).toBe("new line 22");
    expect(saveCustomerAddress.mock.calls[0][0].address3).toBe("new line 23");
    expect(saveCustomerAddress.mock.calls[0][0].address4).toBe("new line 24");
    expect(saveCustomerAddress.mock.calls[0][0].postcode).toBe("new postcode");
    expect(saveCustomerAddress.mock.calls[0][0].customer).toBe(customerAddress.customer);
    expect(component.state('saveInProgress')).toBeTruthy();
});
it("triggers save for a new customer when requested", () => {
    const customerAddress = {};
    const saveCustomerAddress = jest.fn();
    const customerId = 98;

    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
            saveCustomerAddress={saveCustomerAddress}
            customerId={customerId}
        />
    );
    component.instance().handleInputChange("address1_123", 'new line 21');
    component.instance().handleInputChange("address2_123", 'new line 22');
    component.instance().handleInputChange("address3_123", 'new line 23');
    component.instance().handleInputChange("address4_123", 'new line 24');
    component.instance().handleInputChange("postcode_123", 'new postcode');

    component.instance().saveOrCreateCustomerAddress();

    expect(component.state('isChanged')).toBeTruthy();
    expect(saveCustomerAddress.mock.calls.length).toBe(1);
    expect(saveCustomerAddress.mock.calls[0][0].id).toBe(undefined);
    expect(saveCustomerAddress.mock.calls[0][0].address1).toBe("new line 21");
    expect(saveCustomerAddress.mock.calls[0][0].address2).toBe("new line 22");
    expect(saveCustomerAddress.mock.calls[0][0].address3).toBe("new line 23");
    expect(saveCustomerAddress.mock.calls[0][0].address4).toBe("new line 24");
    expect(saveCustomerAddress.mock.calls[0][0].postcode).toBe("new postcode");
    expect(saveCustomerAddress.mock.calls[0][0].customer).toBe(customerId);
    expect(component.state('saveInProgress')).toBeTruthy();
});
it("triggers delete for an existing customer when requested", () => {
    const customerAddress = {
        id: 123,
        address1: "line one",
        address2: "line Ywo",
        address3: "line Three",
        address4: "line Four",
        postcode: "xxxyyy",
        customer: 6
    };
    const deleteCustomerAddress = jest.fn();
    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
            deleteCustomerAddress={deleteCustomerAddress}
        />
    );
    component.instance().onClickDelete();
    expect(deleteCustomerAddress.mock.calls.length).toBe(1);
    expect(deleteCustomerAddress.mock.calls[0][0]).toBe(customerAddress.id);
});
it("clears data for a new customer when delete requested", () => {
    const customerAddress = {};
    const deleteCustomerAddress = jest.fn();
    const customerId = 98;

    const component = shallow(
        <CustomerAddressEdit
            customerAddress={customerAddress}
            deleteCustomerAddress={deleteCustomerAddress}
            customerId={customerId}
        />
    );
    component.instance().handleInputChange("address1_123", 'new line 21');
    component.instance().handleInputChange("address2_123", 'new line 22');
    component.instance().handleInputChange("address3_123", 'new line 23');
    component.instance().handleInputChange("address4_123", 'new line 24');
    component.instance().handleInputChange("postcode_123", 'new postcode');

    component.instance().onClickDelete();
    expect(deleteCustomerAddress.mock.calls.length).toBe(0);
    expect(component.state('address1')).toBe("");
    expect(component.state('address2')).toBe("");
    expect(component.state('address3')).toBe("");
    expect(component.state('address4')).toBe("");
    expect(component.state('postcode')).toBe("");
    expect(component.state('isChanged')).toBeFalsy();
    expect(component.state('isValid')).toBeTruthy();
    expect(component.state('address1Error')).toBe("");
    expect(component.state('postcodeError')).toBe("");
});
