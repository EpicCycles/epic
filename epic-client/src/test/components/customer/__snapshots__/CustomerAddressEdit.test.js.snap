// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a new customer address 1`] = `
<div
  className="grid-row"
  key="rownew"
>
  <EditModelRow
    model={
      Object {
        "customer": 12,
      }
    }
    modelFields={
      Array [
        Object {
          "error": "At least one line of address must be entered",
          "fieldName": "address1",
          "header": "Address",
          "length": 100,
          "required": true,
          "type": "text",
        },
        Object {
          "fieldName": "address2",
          "header": "Line 2",
          "length": 100,
          "type": "text",
        },
        Object {
          "fieldName": "address3",
          "header": "Line 3",
          "length": 100,
          "type": "text",
        },
        Object {
          "fieldName": "address4",
          "header": "Line 4",
          "length": 100,
          "type": "text",
        },
        Object {
          "error": "A country must be selected",
          "fieldName": "country",
          "header": "Country",
          "required": true,
          "type": "country",
        },
        Object {
          "fieldName": "postcode",
          "header": "Postcode",
          "length": 20,
          "type": "text",
          "validator": [Function],
          "validatorAdditionalFields": Array [
            "country",
          ],
        },
      ]
    }
    onChange={[Function]}
    persistedModel={Object {}}
  />
  <div
    className="grid-item"
    key="date_new"
  >
    <span
      id="comment_td_new"
    >
      Add a new address
    </span>
  </div>
  <div
    className="grid-item align_center"
    key="actions_td_new"
  >
    <span
      id="actionsnew"
    />
  </div>
</div>
`;

exports[`renders correctly with a passed customer address 1`] = `
<div
  className="grid-row"
  key="row123"
>
  <EditModelRow
    model={
      Object {
        "address1": "line one",
        "address2": "line Ywo",
        "address3": "line Three",
        "address4": "line Four",
        "customer": 6,
        "id": 123,
        "postcode": "xxxyyy",
      }
    }
    modelFields={
      Array [
        Object {
          "error": "At least one line of address must be entered",
          "fieldName": "address1",
          "header": "Address",
          "length": 100,
          "required": true,
          "type": "text",
        },
        Object {
          "fieldName": "address2",
          "header": "Line 2",
          "length": 100,
          "type": "text",
        },
        Object {
          "fieldName": "address3",
          "header": "Line 3",
          "length": 100,
          "type": "text",
        },
        Object {
          "fieldName": "address4",
          "header": "Line 4",
          "length": 100,
          "type": "text",
        },
        Object {
          "error": "A country must be selected",
          "fieldName": "country",
          "header": "Country",
          "required": true,
          "type": "country",
        },
        Object {
          "fieldName": "postcode",
          "header": "Postcode",
          "length": 20,
          "type": "text",
          "validator": [Function],
          "validatorAdditionalFields": Array [
            "country",
          ],
        },
      ]
    }
    onChange={[Function]}
    persistedModel={
      Object {
        "address1": "line one",
        "address2": "line Ywo",
        "address3": "line Three",
        "address4": "line Four",
        "customer": 6,
        "id": 123,
        "postcode": "xxxyyy",
      }
    }
  />
  <div
    className="grid-item"
    key="date_123"
  >
    <span
      id="comment_td_123"
    >
      Add a new address
    </span>
  </div>
  <div
    className="grid-item align_center"
    key="actions_td_123"
  >
    <span
      id="actions123"
    >
      <Icon
        as="i"
        id="delete-address123"
        name="delete"
        onClick={[Function]}
        title="Delete Address Number"
      />
    </span>
  </div>
</div>
`;
